### PR TITLE
IndexedDB cursor.request

### PR DESCRIPTION
**Pull request by jakearchibald**

### CODE REVIEW

File: IndexedDB/idbcursor-request.htm [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+      .objectStore("my_objectstore")
+      .openKeyCursor();
+
+    req.onsuccess = t.step_func(() => {
+      const cursor = req.result;
+      assert_equals(cursor.request, req, 'cursor.request');
+      assert_readonly(cursor, 'request');
+    });
+  },
+  document.title
```
Comment (inexorabletash) [inline IndexedDB/idbcursor-request.htm]: Remove.

File: IndexedDB/idbcursor-request.htm [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+  (t, db) => {
+    const req = db.transaction("my_objectstore")
+      .objectStore("my_objectstore")
+      .openKeyCursor();
+
+    req.onsuccess = t.step_func(() => {
+      const cursor = req.result;
+      assert_equals(cursor.request, req, 'cursor.request');
+      assert_readonly(cursor, 'request');
+    });
```
Comment (inexorabletash) [inline IndexedDB/idbcursor-request.htm]: Needs `t.done();`

File: IndexedDB/idbcursor-request.htm [hunk 3]
```diff
@@ -0,0 +1,27 @@
+<!DOCTYPE html>
```
Comment (inexorabletash) [inline IndexedDB/idbcursor-request.htm]: Make this a `.any.js` test for more coverage?

Comment (inexorabletash) [reply IndexedDB/idbcursor-request.htm]: ... and remove some boilerplate.

**Review by inexorabletash — CHANGES_REQUESTED** [review]
Please add test cases for:

* IDBCursorWithValue (i.e. non-key cursor)
* Both cursor types with source from index (test here has object store as source)
* Value of cursor.request not within IDB callback and after transaction has ended. (Should still be valid, but one could imagine a non-conforming implementation that needs to break a reference cycle and nulls it out.)

File: IndexedDB/idbcursor-request.htm (hunk 2 above)
Comment (jakearchibald) [reply IndexedDB/idbcursor-request.htm]: Haha I did this right in my run through before cameras were on. Oh well.

**Comment by jakearchibald** [comment]
@inexorabletash done!

File: IndexedDB/idbcursor-request.any.js [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+      let cursor;
+
+      req.onsuccess = t.step_func(() => {
+        cursor = req.result;
+        assert_equals(cursor.request, req, 'cursor.request');
+        assert_readonly(cursor, 'request');
+      });
+
+      req.transaction.oncomplete = t.step_func(() => {
+        setTimeout(t.step_func(() => {
```
Comment (inexorabletash) [inline IndexedDB/idbcursor-request.any.js]: Use `t.step_timeout` per https://web-platform-tests.org/writing-tests/testharness-api.html

File: IndexedDB/idbcursor-request.any.js [hunk 5]
```diff
@@ -0,0 +1,42 @@
...
+          assert_equals(cursor.request, req, 'cursor.request after transaction complete');
+          t.done();
+        }), 0);
+      });
+
+      req.transaction.onerror = t.step_func((event) => {
```
Comment (inexorabletash) [inline IndexedDB/idbcursor-request.any.js]: This can use `t.unreached_func` instead for brevity.

**Review by inexorabletash — CHANGES_REQUESTED** [review]
More nits!

Looking awesome though.

File: IndexedDB/idbcursor-request.any.js (hunk 4 above)
Comment (jakearchibald) [reply IndexedDB/idbcursor-request.any.js]: `step_timeout` is used for "wait a bit" cases, and to allow multipliers to be applied to the duration, whereas I'm using `setTimeout` it to wait until the next task. The timeout is 0 so multipliers don't do anything.

File: IndexedDB/idbcursor-request.any.js (hunk 5 above)
Comment (jakearchibald) [reply IndexedDB/idbcursor-request.any.js]: This is what I get for cargo culting the other tests.

File: IndexedDB/idbcursor-request.any.js (hunk 4 above)
Comment (inexorabletash) [reply IndexedDB/idbcursor-request.any.js]: ack. (maybe we should codify that as `t.step_task` .... )

**Review by inexorabletash — APPROVED** [review]

**Comment by inexorabletash** [comment]
Hmmm, any idea how to get the "Taskcluster (pull_request)" to finish? @Hexcles @foolip

**Comment by foolip** [comment]
@inexorabletash looks like it hasn't even started. @jgraham we've had this issue before in https://github.com/web-platform-tests/wpt/issues/14165 which was then resolved, but now it's happening again. Do you know of ongoing Taskcluster issues?

The easiest way to retrigger all CI systems is to close and reopen a PR, so I'll do that. If Taskcluster still doesn't start, we should just admin merge.

**Comment by foolip** [comment]
This is concerning, there's still no "Details" link for Taskcluster, so it hasn't told GitHub about any ongoing run. @jgraham, absent that, is there any way to figure out if Taskcluster started?

**Comment by inexorabletash** [comment]
Admin-merge pls?

**Comment by Hexcles** [comment]
Given that even the close-reopen trick didn't work, I'm admin-merging this. Debugging the infra shouldn't block developers' work.

**Pull request merged**